### PR TITLE
META-3067 :- Add classification displayName in validation msg

### PR DESCRIPTION
**Pull request by nixon-atlan**
## Change description
Delete a Classification having assets associated with it. The validation message that is displayed at the top has incorrect text.
[Linear](https://linear.app/atlanproduct/issue/META-3067/classification-the-validation-message-isnt-correct-when-we-try)
> Description here

## Type of change
- [ ] Bug fix (fixes an issue)
- [ ] New feature (adds functionality)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] Application changes have been tested thoroughly
- [ ] Automated tests covering modified code pass

### Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### Code review 

- [ ] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [ ] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [ ] Pull request linked to task tracker where applicable

### CODE REVIEW

**Review by mehtaanshul — APPROVED** [review]

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v2/AtlasClassificationDefStoreV2.java [hunk 1]
```diff
@@ -336,7 +337,8 @@ public AtlasVertex preDeleteByGuid(String guid) throws AtlasBaseException {
         String typeName = AtlasGraphUtilsV2.getEncodedProperty(ret, TYPENAME_PROPERTY_KEY, String.class);
 
         if (AtlasGraphUtilsV2.typeHasInstanceVertex(typeName)) {
-            throw new AtlasBaseException(AtlasErrorCode.TYPE_HAS_REFERENCES, typeName);
+            String displayName = existingDef.getDisplayName() != null ? existingDef.getDisplayName() : "";
```
Comment (mehtaanshul) [inline repository/src/main/java/org/apache/atlas/repository/store/graph/v2/AtlasClassificationDefStoreV2.java:340]: Null check on existingDef

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v2/AtlasClassificationDefStoreV2.java [hunk 2]
```diff
@@ -305,7 +305,8 @@ public AtlasVertex preDeleteByName(String name) throws AtlasBaseException {
         AtlasVertex ret = typeDefStore.findTypeVertexByNameAndCategory(name, TypeCategory.TRAIT);
 
         if (AtlasGraphUtilsV2.typeHasInstanceVertex(name)) {
-            throw new AtlasBaseException(AtlasErrorCode.TYPE_HAS_REFERENCES, name);
+            String displayName = existingDef.getDisplayName() != null ? existingDef.getDisplayName() : "";
```
Comment (mehtaanshul) [inline repository/src/main/java/org/apache/atlas/repository/store/graph/v2/AtlasClassificationDefStoreV2.java:308]: Null check on existingDef

**Review by mehtaanshul — CHANGES_REQUESTED** [review]
Added a comment

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v2/AtlasClassificationDefStoreV2.java (hunk 1 above)
Comment (nixon-atlan) [reply repository/src/main/java/org/apache/atlas/repository/store/graph/v2/AtlasClassificationDefStoreV2.java]: added NPE check for existingDef

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v2/AtlasClassificationDefStoreV2.java (hunk 2 above)
Comment (nixon-atlan) [reply repository/src/main/java/org/apache/atlas/repository/store/graph/v2/AtlasClassificationDefStoreV2.java]: added NPE check for existingDef

**Pull request closed without merging**